core/src/lib/auth/providers/userapikey/CoreUserAPIKeyAuthProviderClient.ts

```diff
@@ -19,150 +19,149 @@ enum ApiKeyFields {
   ) {
-    let baseRoute = `${authRoutes.baseAuthRoute}/api_keys`
-    super(UserAPIKeyAuthProvider.TYPE, requestClient, baseRoute)
+    let baseRoute = `${authRoutes.baseAuthRoute}/api_keys`;
+    super(UserAPIKeyAuthProvider.TYPE, requestClient, baseRoute);
   }
 
   /**
    * Creates a user API key that can be used to authenticate as the current user.
-   * 
+   *
    * @param name the name of the API key to be created.
    */
-  protected createApiKey(name: string): Promise<UserAPIKey> {
+  public createApiKey(name: string): Promise<UserAPIKey> {
     const reqBuilder = new StitchAuthDocRequest.Builder();
-    reqBuilder
-      .withMethod(Method.POST)
-      .withPath(this.baseRoute);
+    reqBuilder.withMethod(Method.POST).withPath(this.baseRoute);
     reqBuilder
       .withDocument({
         [ApiKeyFields.NAME]: name
       })
       .withRefreshToken();
-    
+
     return this.requestClient
-      .doAuthenticatedJSONRequestRaw(reqBuilder.build())
+      .doAuthenticatedRequest(reqBuilder.build())
       .then(response => {
-        return UserAPIKey.readFromAPI(response.body)
+        return UserAPIKey.readFromAPI(response.body!);
       })
-      .catch(err => { throw StitchError.wrapDecodingError(err) })
+      .catch(err => {
+        throw StitchError.wrapDecodingError(err);
+      });
   }
 
   /**
    * Fetches a user API key associated with the current user.
-   * 
+   *
    * @param keyId the id of the API key to fetch.
    */
-  protected fetchApiKey(keyId: ObjectID): Promise<UserAPIKey> {
+  public fetchApiKey(keyId: ObjectID): Promise<UserAPIKey> {
     const reqBuilder = new StitchAuthRequest.Builder();
     reqBuilder
       .withMethod(Method.GET)
       .withPath(this.getApiKeyRoute(keyId.toHexString()));
-    reqBuilder
-      .withRefreshToken();
-    
+    reqBuilder.withRefreshToken();
+
     return this.requestClient
       .doAuthenticatedRequest(reqBuilder.build())
       .then(response => {
-        return UserAPIKey.readFromAPI(response.body)
+        return UserAPIKey.readFromAPI(response.body!);
       })
-      .catch(err => { throw StitchError.wrapDecodingError(err) })
+      .catch(err => {
+        throw StitchError.wrapDecodingError(err);
+      });
   }
 
   /**
    * Fetches the user API keys associated with the current user.
    */
   protected fetchApiKeys(): Promise<UserAPIKey[]> {
     const reqBuilder = new StitchAuthRequest.Builder();
-    reqBuilder
-      .withMethod(Method.GET)
-      .withPath(this.baseRoute);
-    reqBuilder
-      .withRefreshToken();
-    
+    reqBuilder.withMethod(Method.GET).withPath(this.baseRoute);
+    reqBuilder.withRefreshToken();
+
     return this.requestClient
       .doAuthenticatedRequest(reqBuilder.build())
       .then(response => {
-        if(Array.isArray(response.body)) {
-          const keys = Array.from(response.body)
-          return keys.map(value => UserAPIKey.readFromAPI(value))
+        const json = JSON.parse(response.body!);
```

Comment:
[minor] This should ideally be using the `doAuthenticatedRequest` with the simple `Decoder<T>` we discussed.

Reply:
Same thing applies to other PRs here.